src/main/java/de/uhd/ifi/se/decision/management/jira/persistence/CodeClassKnowledgeElementPersistenceManager.java

```diff
@@ -0,0 +1,264 @@
+import net.java.ao.Query;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.diff.DiffEntry;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CodeClassKnowledgeElementPersistenceManager extends AbstractPersistenceManagerForSingleLocation {
+	private static final Logger LOGGER = LoggerFactory.getLogger(JiraIssueTextPersistenceManager.class);
+	private static final ActiveObjects ACTIVE_OBJECTS = ComponentGetter.getActiveObjects();
+
+	public CodeClassKnowledgeElementPersistenceManager(String projectKey) {
+		this.projectKey = projectKey;
+		this.documentationLocation = DocumentationLocation.COMMIT;
+	}
+
+	@Override
+	public boolean deleteDecisionKnowledgeElement(long id, ApplicationUser user) {
+		if (id <= 0 || user == null) {
+			LOGGER.error(
+					"Element cannot be deleted since it does not exist (id is less than zero) or the user is null.");
+			return false;
+		}
+		boolean isDeleted = false;
+		for (CodeClassElementInDatabase databaseEntry : ACTIVE_OBJECTS.find(CodeClassElementInDatabase.class,
+				Query.select().where("ID = ?", id))) {
+			GenericLinkManager.deleteLinksForElement(id, DocumentationLocation.COMMIT);
+			KnowledgeGraph.getOrCreate(projectKey).removeVertex(new KnowledgeElementImpl(databaseEntry));
+			isDeleted = CodeClassElementInDatabase.deleteElement(databaseEntry);
+		}
+		return isDeleted;
+	}
+
+	@Override
+	public KnowledgeElement getDecisionKnowledgeElement(long id) {
+		KnowledgeElement element = null;
+		for (CodeClassElementInDatabase databaseEntry : ACTIVE_OBJECTS.find(CodeClassElementInDatabase.class,
+				Query.select().where("ID = ?", id))) {
+			element = new KnowledgeElementImpl(databaseEntry);
+		}
+		return element;
+	}
+
+	@Override
+	public KnowledgeElement getDecisionKnowledgeElement(String key) {
+		String id = key.split("-")[1];
+		return getDecisionKnowledgeElement(id);
+	}
+
+	public KnowledgeElement getDecisionKnowledgeElement(KnowledgeElement element) {
+		if (element == null || element.getId() <= 0) {
+			return null;
+		}
+		return this.getDecisionKnowledgeElement(element.getId());
+	}
+
+	public KnowledgeElement getDecisionKnowledgeElementByNameAndIssueKeys(String name, String issueKeys) {
+		KnowledgeElement element = null;
+		for (CodeClassElementInDatabase databaseEntry : ACTIVE_OBJECTS.find(CodeClassElementInDatabase.class,
+				Query.select().where("FILE_NAME = ? AND JIRA_ISSUE_KEYS = ?", name, issueKeys))) {
+			element = new KnowledgeElementImpl(databaseEntry);
+		}
+		return element;
+	}
+
+	public CodeClassElementInDatabase getEntryForKnowledgeElement(KnowledgeElement element) {
+		Long id = element.getId();
+		for (CodeClassElementInDatabase databaseEntry : ACTIVE_OBJECTS.find(CodeClassElementInDatabase.class,
+				Query.select().where("ID = ?", id))) {
+			return databaseEntry;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using a branching statement as the last in a loop.](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=5229259)